tests/web/test_web_client.py

```diff
@@ -18,11 +19,34 @@ def setUp(self):
     def tearDown(self):
         pass
 
+    pattern_for_language = re.compile("python/(\\S+)", re.IGNORECASE)
+    pattern_for_package_identifier = re.compile("slackclient/(\\S+)")
+
     def test_api_calls_return_a_response_when_run_in_sync_mode(self, mock_request):
         resp = self.client.api_test()
         self.assertFalse(asyncio.isfuture(resp))
         self.assertTrue(resp["ok"])
 
+    def test_api_calls_include_user_agent(self, mock_request):
+        self.client.api_test()
+        mock_call_kwargs = mock_request.call_args[1]
+        self.assertIn("req_args", mock_call_kwargs)
+        mock_call_req_args = mock_call_kwargs["req_args"]
+        self.assertIn("headers", mock_call_req_args)
+        mock_call_headers = mock_call_req_args["headers"]
+        self.assertIn("User-Agent", mock_call_headers)
+        mock_call_user_agent = mock_call_headers["User-Agent"]
```

Comment:
unpacking each property of the nested value one-by-one to avoid trying to lookup named members of `None` (in case one doesn't exist). i figured asserting along the way is good so that if the test fails, the failure message indicates exactly which lookup failed. open to feedback for a cleaner way to accomplish this!